include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -271,8 +303,90 @@ namespace RINGMesh
 
         GEO::AttributesManager attributes_manager_;
     };
+    ALIAS_2D_AND_3D( CartesianGridBase );
+
+    template < index_t DIMENSION >
+    class CartesianGrid final : public CartesianGridBase< DIMENSION >
+    {
+        friend class CartesianGridBuilder< DIMENSION >;
+    };
     ALIAS_2D_AND_3D( CartesianGrid );
 
+    template <>
+    class CartesianGrid< 3 > final : public CartesianGridBase< 3 >
+    {
+        friend class CartesianGridBuilder< 3 >;
+
+    public:
+        CartesianGrid( ivec3 nb_cells_in_each_direction,
+            ReferenceFrame3D vec_cartesian_axis )
+            : CartesianGridBase(
+                  nb_cells_in_each_direction, vec_cartesian_axis ),
+              grid_cage_()
+        {
+            vec3 highest_coordinates_point{
+                cartesian_frame_.origin()
+                + cartesian_frame_[0] * nb_cells_axis( 0 )
+                + cartesian_frame_[1] * nb_cells_axis( 1 )
+                + cartesian_frame_[2] * nb_cells_axis( 2 )
+            };
+            for( auto i : range( 3 ) )
+            {
+                grid_cage_.emplace_back(
+                    cartesian_frame_[i], cartesian_frame_.origin() );
+                grid_cage_.emplace_back(
+                    cartesian_frame_[i], highest_coordinates_point );
+            }
+        }
+
+        const std::vector< Geometry::Plane >& grid_cage() const
+        {
+            return grid_cage_;
+        }
+
+    private:
+        /// The 6 planes of the grid cage are ordered in this way :
+        /// First the 2 with a normal to the first axis of the grid,
+        /// then the 2 with a normal to the second axis of the grid,
+        /// and finally the 2 with a normal to the third axis of the grid.
+        /// Each time, the first of the 2 planes in question is the one which
+        /// contains the origin of the grid, and the second is the one which
+        /// contains the point with the highest local coordinates in the grid.
+        std::vector< Geometry::Plane > grid_cage_;
+    };
+
+    template <>
+    class CartesianGrid< 2 > final : public CartesianGridBase< 2 >
+    {
+        friend class CartesianGridBuilder< 2 >;
+
+    public:
+        CartesianGrid( ivec2 nb_cells_in_each_direction,
+            ReferenceFrame2D vec_cartesian_axis )
+            : CartesianGridBase(
+                  nb_cells_in_each_direction, vec_cartesian_axis )
+        {
+        }
+
+        const std::vector< Geometry::Segment2D > grid_cage() const
+        {
+            std::vector< Geometry::Segment2D > cage;
+            cage.resize( 4 );
+            cage.emplace_back( cartesian_frame_.origin(),
```

Comment:
Don't you find it clearer this way?

Reply:
Yes, less characters and easier to read.